src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -830,14 +842,9 @@ private function filterRelativeXPath($xpath)
 
         $crawler = $this->createSubCrawler(null);
 
-        foreach ($this->nodes as $node) {
+        foreach ($this as $node) {
```

Comment:
this change should be reverted. There is no reason to iterate on the crawler rather than iterator directly on the internal array